moped-database/migrations/1700149643858_parent_audit_values/up.sql

```diff
@@ -0,0 +1,69 @@
+-- Altering moped_project table to add a new column for tracking the user who last updated a record
+ALTER TABLE moped_project ADD COLUMN updated_by_user_id INTEGER DEFAULT NULL;
+COMMENT ON COLUMN moped_project.updated_by_user_id IS 'User ID of the person who last updated the project';
+
+-- Creating or replacing a function to update audit logs in parent records
+CREATE OR REPLACE FUNCTION update_parent_records_audit_logs()
+RETURNS TRIGGER AS $$
+DECLARE
+  project_id_variable INTEGER;
+  user_id_variable INTEGER;
+BEGIN
+  -- Select the user ID who last updated the project component
+  SELECT updated_by_user_id INTO user_id_variable
+  FROM moped_proj_components
+  WHERE project_component_id = NEW.component_id;
```

Comment:
if i'm following, this is always setting the `moped_proj_components. updated_by_user_id` to its previous value, 
whereas we need to be pulling the `updated_by_user_id` value of the current user has made the edits. i didn't write this in scope, but we might need to add audit fields to the `feature_` tables so that we have a current user ID to assign to this column?

Reply:
John, I may have put a seriously dim-witted comment in there, but I'm going to check / fix it to make sure it's actually representing the current user's ID and not whomever last made an edit on the moped_proj_component's record. I did most of my testing by creating a new component, and I may have massively overlooked this because all of my records were created / edited and whatnot by user_id 1. Thanks for flagging this!! 🙏

Reply:
John -- first off - your comment makes perfect sense. I was pulling the last editor field off the `moped_proj_components` instead of off the feature tables which contained the edited or created data. I've absolutely corrected that with a dynamically constructed query in my function and hooked it up to triggers for the feature tables. This is so absolutely the right way to do this, and I really appreciate you noting this monstrously hair-brained oversight on my part.

I'm excited to ask for your review again -- I think I'm back on track here.